Automattic/jetpack
# Refactor carousel to use swiper.js

Fixes #15678 
Fixes #6069 
Fixes #18923 
Fixes #9011 


Currently the touch screen usability of the carousel module is not as good as other online galleries, or the Jetpack slideshow block.

This PR removes the custom carousel functionality and touch screen utils and replaces them with the 3rd party [swiper.js](https://swiperjs.com/) library which is used by the Jetpack slideshow block.

![2021-06-24 20 34 15](https://user-images.githubusercontent.com/1464705/123365698-a2688180-d52b-11eb-831c-1dd1f9ab9adc.gif)


#### Testing instructions:
1. Checkout this PR in local jetpack dev env
2. Enable the Jetpack carousel option in jetpack settings
2. Add a gallery to a page or post, load in front end and check that the carousel lightbox loads and works as expected
3. On a mobile device run the gallery and check that swipe works well, and also pinch zoom and pan

#### Ensure existing Carousel features and toggles work as expected
1. Go to `/wp-admin/admin.php?page=jetpack#/writing` and toggle showing exif and comments (each setting should show/hide icons in the bottom right of the carousel)
2. From the settings above, switch to the White carousel color scheme and test that it works correctly

#### Does this pull request change what data or activity we track or use?
No
